### PR TITLE
chore(github-comments): add success/fail metrics for comment task

### PR DESCRIPTION
**Pull request by cathteng**
If getting the comment returns a 404 (the comment doesn't exist), emit a metric and continue in the loop rather than capture the exception. Also emit a metric if the GET succeeds.

For ER-1745

### CODE REVIEW

File: src/sentry/tasks/integrations/github/pr_comment.py [hunk 1]
```diff
@@ -309,6 +309,12 @@ def github_comment_reactions():
                 metrics.incr("github_pr_comment.comment_reactions.rate_limited_error")
                 break
 
+            if e.code == 404:
+                metrics.incr("github_pr_comment.comment_reactions.not_found_error")
+                continue
+
             metrics.incr("github_pr_comment.comment_reactions.api_error")
             sentry_sdk.capture_exception(e)
             continue
```
Comment (schew2381) [inline src/sentry/tasks/integrations/github/pr_comment.py:318]:
```suggestion
            if e.code == 404:
                metrics.incr("github_pr_comment.comment_reactions.not_found_error")
            else:
                metrics.incr("github_pr_comment.comment_reactions.api_error")
                sentry_sdk.capture_exception(e)
            continue
```

File: src/sentry/tasks/integrations/github/pr_comment.py [hunk 2]
```diff
@@ -309,6 +309,12 @@ def github_comment_reactions():
                 metrics.incr("github_pr_comment.comment_reactions.rate_limited_error")
                 break
 
+            if e.code == 404:
```
Comment (schew2381) [inline src/sentry/tasks/integrations/github/pr_comment.py:312]: What other error codes can be thrown from here?

Comment (cathteng) [reply src/sentry/tasks/integrations/github/pr_comment.py:312]: the one we want to check for is if the comment has been deleted (presumably) which is why the API is returning 404 not found. this is happening quite frequently so we are checking for this one specifically and not capturing the exception, but for other errors we will still capture the exception

File: src/sentry/tasks/integrations/github/pr_comment.py (hunk 1 above)
Comment (cathteng) [reply src/sentry/tasks/integrations/github/pr_comment.py:318]: thanks!

**Review by AniketDas-Tekky — APPROVED** [review]

**Pull request merged**